File: src/main/scala/es/us/idea/dmn4spark/diagnosis/graph/adapters/VertexIDProvider.java
package es.us.idea.dmn4spark.diagnosis.graph.adapters;

import es.us.idea.dmn4spark.diagnosis.graph.components.basic.Vertex;
import org.jgrapht.ext.VertexNameProvider;

public class VertexIDProvider implements VertexNameProvider<Vertex> {

    @Override
    public String getVertexName(Vertex vertex) {
        //return vertex.toString().split("@")[1].split("\\[")[0];
        return "A"+vertex.toString().split("@")[1].split("\\[")[0];
    }
}

File: src/main/scala/es/us/idea/dcdq/diagnosis/graph/adapters/VertexIDProvider.scala
package es.us.idea.dcdq.diagnosis.graph.adapters

import es.us.idea.dcdq.diagnosis.graph.components.basic.Vertex
import org.jgrapht.ext.VertexNameProvider


class VertexIDProvider extends VertexNameProvider[Vertex] {
  override def getVertexName(vertex: Vertex): String = { //return vertex.toString().split("@")[1].split("\\[")[0];
    "A" + vertex.toString.split("@")(1).split("\\[")(0)
  }
}
